### PR TITLE
Feature/13140 add permissions management for wallet connect 

### LINKED ISSUE #13140: Add permissions management for Wallet Connect
# Feature Issue

[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to be able to revoke access for Wallet Connect to my Status account from `Privacy and security` settings **so that** I don't need to do it in a desktop DApp. 

### Description

[comment]: # (Describe the feature you would like and include designs and Figma links if available.)
[comment]: # (Describe the use cases the implementation should cover. E.g. New users, existing users, public chat, group chat, etc)
*Summary*: Currently, we can't revoke Wallet Connect's access to the Status account through the app's settings as we can with DApp accesses (`Profile` > `Privacy and security` > `Set DApp access permissions`). To do this we need to open a desktop version of DApp and disconnect the wallet there.

### Acceptance Criteria

Permissions management for Wallet Connect is added to Status settings

### Notes

Improvement for [Wallet Connect 1.0](https://github.com/status-im/status-react/pull/13032)

### PR DESCRIPTION
**Pull request by siddarthkay**
fixes #13140 

### Review notes
PR to implement disconnecting an already connected wallet via wallet connect 

### Testing notes
- connect a wallet via walletconnect by scanning the QR code.
- you can now click on the three dots on the top right hand side of the wallet screen and find a new menu item added to the bottom popover sheet called "Manage app connections"
- on tap of that option you get the popover to disconnect the connection.

#### Platforms

- Android
- iOS

#### Areas that maybe impacted
wallet connect 

##### Functional
- wallet / transactions
- dapps / app browsing

##### Non-functional
may impact some network consumption

status: ready

### CODE REVIEW

**Comment by Parveshdhull** [comment]
hi @siddarthkay , please mark PR WIP also in title

**Comment by flexsurfer** [comment]
hey @siddarthkay any updates? this PR is a blocker for the next release

**Comment by siddarthkay** [comment]
Hi @flexsurfer : I have achieved the desired functionality in status-go and currently working on testing it with status-react.
This is very close to completion.I Shall push the final code this week and assign it for review.

**Comment by siddarthkay** [comment]
https://user-images.githubusercontent.com/64726664/182362884-b3ac1659-f5a0-4a4b-abe5-83005eab0ad4.mov

we can now disconnect a wallet session.

Only case it does not work is when you sync sessions across devices.. that is a work in progress.
Can that be made as a separate PR and is the current progress good enough for this milestone 1.20?

@cammellos  @qoqobolo

**Comment by qoqobolo** [comment]
> Only case it does not work is when you sync sessions across devices.. that is a work in progress.
Can that be made as a separate PR and is the current progress good enough for this milestone ?

@siddarthkay Thanks again for your work!

And for the record, as we agreed on the planning call it would be okay to implement the syncing part separately as long as it doesn't introduce other issues (will be checked during testing).

File: translations/en.json [hunk 1]
```diff
@@ -1754,8 +1755,9 @@
     "contact-request-pending": "Pending...",
     "removed-from-contacts": "Removed from contacts",
     "mutual-contact-requests": "Mutual contact requests",
-    "pending": "Pending",
     "negative": "Negative",
     "positive": "Positive",
-    "public": "Public"
+    "public": "Public",
+    "app-connections": "App connections",
```
Comment (flexsurfer) [inline translations/en.json]: app-connections duplicated

File: src/status_im/ui/screens/wallet/accounts/styles.cljs [hunk 2]
```diff
@@ -48,6 +48,7 @@
    :left            0})
 
 (def card-margin 8)
+
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/accounts/styles.cljs]: could you revert this file

File: translations/en.json (hunk 1 above)
Comment (siddarthkay) [reply translations/en.json]: removed duplicate key, again sorry for missing this out after rebasing develop

File: src/status_im/ui/screens/wallet/accounts/styles.cljs (hunk 2 above)
Comment (siddarthkay) [reply src/status_im/ui/screens/wallet/accounts/styles.cljs]: Done!

**Comment by flexsurfer** [comment]
video demo doesn't look like production ready ? don't we have designs ?

**Comment by siddarthkay** [comment]
> video demo doesn't look like production ready ? don't we have designs ?


The Daap used for demo video is 1inch.io and its Image asset does not load properly and it may give you an impression that something is incomplete but it is not. 
I have used the existing / approved components to showcase the connections popover.
Perhaps this new video would showcase that below, here I connect session of zerion Daap.


https://user-images.githubusercontent.com/64726664/182658449-5ef2ef2c-bf7a-4147-bde1-7ce25e1f7866.mov

**Comment by siddarthkay** [comment]
@flexsurfer : Here are the designs https://www.figma.com/file/UQSs6JFVpYRhpgCimvVDPl/Swaps%2C-WalletConnect-%F0%9F%94%926-Dec-21?node-id=111%3A16654

Do you still concerns with the implementation? If yes can you please elaborate.

**Review by flexsurfer — APPROVED** [review]

**Comment by qoqobolo** [comment]
@siddarthkay

### ISSUE 1: The `Manage app connections` menu item disappears after relogin
Although the wallet is still connected in a DApp.
Thus, you can no longer manage your app connections after relogin.
Not sure if it's expected, since other DApp permissions (in Profile > Privacy and security > Set DApp access permissions) keep displaying after relogin. 

**Steps:**
1. Connect a wallet via walletconnect (tested on https://app.uniswap.org/)
2. Re-login
3. Open Wallet > open the three-dots menu

**OS:** Android, iOS

https://user-images.githubusercontent.com/67952253/183387777-7b0f65e2-b138-41ca-acb9-24702e9eb33c.mp4

___

Also, can I ask you to rebase this PR on the latest develop, please?

**Comment by siddarthkay** [comment]
> Also, can I ask you to rebase this PR on the latest develop, please?

Will Rebase on latest develop!

**Review by briansztamfater — APPROVED** [review]

**Comment by siddarthkay** [comment]
@qoqobolo : I have identified the fix for issue 1 and I am currently working on it. 
I will ping here once the fix is ready for testing.

Thanks for reporting.

**Comment by qoqobolo** [comment]
@siddarthkay thanks for the update!

**Comment by cammellos** [comment]
Closing in favor of https://github.com/status-im/status-mobile/pull/13799

**Pull request closed without merging**